src/c++/perf_analyzer/test_load_manager_base.h

```diff
@@ -103,29 +127,32 @@ class TestLoadManagerBase {
     // there are never any overlapping requests -- they always immediately exit
     //
     if (params_.sequence_length != 1) {
-      CHECK(
-          params_.num_of_sequences ==
-          stats->sequence_status.max_live_seq_count);
+      expected_num_seq = std::min(expected_num_seq, params_.sequence_id_range);
+      CHECK(expected_num_seq == stats->sequence_status.max_live_seq_count);
     }
 
     // Make sure that the length of each sequence is as expected
-    // (The code explicitly has a 20% slop, so that is what we are checking)
     //
-    auto num_sequences = params_.num_of_sequences;
+    // All but X of them should be within 20% (The code explicitly has a 20%
+    // slop) of the requested sequence length, where X is the number of
+    // sequences (This is due to the shutdown of sequences at the end that will
+    // create shorter than expected sequences)
+    //
     auto num_values = stats->sequence_status.seq_lengths.size();
+    auto max_len = params_.sequence_length * 1.2;
+    auto min_len = params_.sequence_length * 0.8;
+    auto num_allowed_to_be_below_min_len = expected_num_seq;
+    auto num_below_min_len = 0;
```

Comment:
`num_allowed_to_be_below_min_len`

`num_below_min_len`

number of _what_? and can you explain why there allowed to be below whatever it is? and why there are that many below?

Reply:
Ah, it looks like number of sequences. And the number allowed to be below minimum length is equal to the number of sequences because this represents the "last pass" where the rate was changed?

Reply:
This is checking the length of the sequences. It was explained in the comment above.

Every sequence that executes to completion will be within [20%](https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/load_manager.cc#L880) of the requested sequence length. This checks that, except that all of the "finish this sequence early" code can obviously lead to some sequences being shorter than that because they are terminated early. This also allows X to be below the minimum length, where X is the number of sequences.